### DISCUSSION TITLE
kql cf-cache-status: DYNAMIC

### DISCUSSION THREAD

**Discussion by Jorckx** [johannschopplich/nuxt-kql#25] [Q&A] [answered]
Hi,

As a first time Cloudflare user I want to leverage the greatness of their cache-layer. For most of my content I get to see that the cf-chache-status is 'HIT'. However, only the Kql data seems to be be set to 'DYNAMIC'.

I've already set up a hook in kibry to purge the cache when a page is updated/created/removed.

I've been looking to many sources, but nothing that really helped me. I was wondering if you have a solution for this? The fetched data is pretty extensive sometimes… so the loading times are a bit of a pain to be honest.

**Comment by Jorckx** [comment]
or is it in this case better to use pnpm run generate as build command on Cloudflare? I'm is that supported with this module?

**Comment by johannschopplich** [comment, marked as answer]
That depends on your usecase. You can combine both.

I recommend you read the Nuxt KQL [Server Side Caching Guide](https://nuxt-kql.byjohann.dev/config/caching.html#server-side-caching). It explains how to use Cloudflare KV storage as the caching layer. Nuxt natively supports it, since its server Nitro uses [unstroage](https://unstorage.unjs.io).

Of course you can also use `pnpm run generate`. Nuxt KQL provides a data fetching layer. If you want to pre-generate pages at build-time, it's completely up to you to do so. 

For my projects, I generatlly tend to use the KV storage from Cloudflare, but for other projects with control over the build chain it makes sense to just pre-render the whole page and deploy to Cloudflare Pages. Check out my [Cacao Kit](https://github.com/johannschopplich/cacao-kit-frontend/), which is a best-practice Kirby Nuxt integration. The [build pipeline](https://github.com/johannschopplich/cacao-kit-frontend/blob/main/.github/workflows/deploy.yml) of the repo build a server, but you can switch that to static generation easily.

**Reply by Jorckx** [reply]
For this case I specifically want it to be a static to have the highest possible pagespeeds. But If this doesnt work I would opt into the kv storage sollution… I've tried both without anything changing somehow.

https://muce.be/coming-soon/
I've set it up with pnpm run generate, but as you can see it still has the {kql} objects in the network tab.

my config looks as followed:

```

  …
  kql: {
    auth: 'bearer',
    prefetch: {
      kirbySite: siteQuery,
    },    
  },
…
  nitro: {
    experimental: {
      openAPI: false,
    },
},
…

``` 

When i add `ssr: false` the deployment is exacuted. on `true` it somehow trows a build error…

I feel like I tried everything, but find it difficult to understand what's happening.

**Reply by Jorckx** [reply]
When using the storage I don't seem to be able to set the binding to point to the cloudflare kv which is set to __MUCE_KQL.
Do i need to provide the id of that kv? as shown in these docs: https://unstorage.unjs.io/drivers/cloudflare

None of the data is written to the kv when I only set it up like this:
```
  nitro: {
    storage: {
      kql: {
        driver: 'cloudflareKVBinding',
        binding: '__MUCE_KQL',
      },
    },
    devStorage: {
      kql: {
        driver: 'fs',
        base: '.data',
      },
    },
  },

```

**Reply by johannschopplich** [reply]
Yes, you need to set up a KV storage in the Cloudflare dashboard and add the binding + ID to your `wrangler.toml` (if deployed with `wrangler deploy`). Then, you can reference the binding name in your Nitro config.

Unfortunately, I don't have the resources to provide more free help. If you need help setting up KV storage, feel free to book some consulting hours with me. 🙂
